Tests/tests/test_FrontPanel.cpp

```diff
@@ -0,0 +1,724 @@
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("powerLedOn")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("powerLedOff")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setClockBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getClockBrightness")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getFrontPanelLights")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getPreferences")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setPreferences")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setLED")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setBlink")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("set24HourClock")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("is24HourClock")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setClockTestPattern")));
+}
+
+TEST_F(FrontPanelInitializedEventDsTest, setBrightnessWIndex)
+{
+
+    ON_CALL(frontPanelIndicatorMock, getInstanceString)
+        .WillByDefault(::testing::Invoke(
+            [&](const std::string& name) -> device::FrontPanelIndicator& {
+                EXPECT_EQ("Power", name);
+                return device::FrontPanelIndicator::getInstance();
+            }));
+
+        
+    ON_CALL(frontPanelIndicatorMock, getName())
+        .WillByDefault(::testing::Return("Power"));
+
+
+    EXPECT_CALL(frontPanelIndicatorMock, setBrightness(::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](int brightness, bool toPersist) {
+                EXPECT_EQ(brightness, 1);
+            }));
+
+    EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("setBrightness"), _T("{\"brightness\": 1,\"index\": \"power_led\"}"), response));
+    EXPECT_EQ(response, string("{\"success\":true}"));
+}
+TEST_F(FrontPanelInitializedEventDsTest, setBrightnessClock)
+{
+    ON_CALL(frontPanelTextDisplayMock, getInstanceByName)
+        .WillByDefault(::testing::Invoke(
+            [&](const std::string& name) -> device::FrontPanelTextDisplay& {
+                EXPECT_EQ("Text", name);
+                return device::FrontPanelTextDisplay::getInstance();
+            }));
+
+
+
+    ON_CALL(frontPanelConfigImplMock, getIndicators())
+        .WillByDefault(::testing::Return(device::List<device::FrontPanelIndicator>({device::FrontPanelIndicator::getInstance()})));
+    ON_CALL(frontPanelIndicatorMock, getName())
+        .WillByDefault(::testing::Return("red"));
+
+    ASSERT_TRUE(dsFrontPanelModeChange != nullptr);
+
+    IARM_Bus_PWRMgr_EventData_t eventData;
+    eventData.data.state.newState =IARM_BUS_PWRMGR_POWERSTATE_ON;
+    eventData.data.state.curState =IARM_BUS_PWRMGR_POWERSTATE_STANDBY;
+
+    handler.Subscribe(0, _T("powerModeChange"), _T("client.events.powerModeChange"), message);
+
+    dsFrontPanelModeChange(IARM_BUS_PWRMGR_NAME, IARM_BUS_PWRMGR_EVENT_MODECHANGED, &eventData , 0);
```

Comment:
how does theis dsFrontPanelModeChange thing relate to the test?

why subscribe to "powerModeChange"?

Reply:
@npoltorapavlo 

```
#pragma once

#include <gmock/gmock.h>

#include "FrontPanelIndiciatorMock.h"

class FrontPanelTextDisplayMock : public device::FrontPanelTextDisplayImpl, public FrontPanelIndicatorMock {
public:
    virtual ~FrontPanelTextDisplayMock() = default;
    MOCK_METHOD(device::FrontPanelTextDisplay&, getInstanceByName, (const std::string& name), (override));
    MOCK_METHOD(device::FrontPanelTextDisplay&, getInstanceById, (int id), (override));
    MOCK_METHOD(int,getCurrentTimeFormat, (), (const, override));
    MOCK_METHOD(void,setTimeFormat, (const int iTimeFormat), (const, override));
    MOCK_METHOD(int,getTextBrightness, (), (const, override));
    MOCK_METHOD(void,setTextBrightness, (const int brightness), (const, override));
    MOCK_METHOD(void,setText, (const std::string text), (const, override));
    MOCK_METHOD(void,setMode, (const int mode), (const, override));
};
```

Is what the function looks like when I swap it over, and when I run a clean build I get the 'following functions are pure within' error message.

Reply:
@npoltorapavlo You were right, the subscribe calls were unneeded, and have been removed.